velox/type/Filter.h

```diff
@@ -297,81 +299,84 @@ class AlwaysFalse final : public Filter {
 
   static FilterPtr create(const folly::dynamic& /*obj*/);
 
-  bool testingEquals(const Filter& other) const final {
+  bool testingEquals(const Filter& other) const override {
```

Comment:
`final` is important for inlining, we should keep all the `final`s

Reply:
@Yuhta We have the same goal of keeping high perf by avoiding virtual table lookup for a override method.

In C++, a class marked as `final` implies its override methods are all `final`. [1] [2]
Since all the derived classes are marked as `final`, we are good here.

On the other hand, it is preferred to mark a derived method `override` in a derived class, so that the compiler knows it is overriding a virtual method from the base class, instead of adding a new method in the derived class (i.e., in a rare case of a typo in the method name).

What do you think then?

[1]: https://devblogs.microsoft.com/cppblog/the-performance-benefits-of-final-classes/
[2]: https://quuxplusone.github.io/blog/2021/02/15/devirtualization/#table-of-results

Reply:
Either `final` or `override` is mandatory on override method, you cannot merge code without them. I would advocate for `final` on method need inlining, because we only need to check the method signature to make sure it's inlinable, instead of go up to check if the class is final or not.  There is also a chance that the class `final` got removed during refactoring and break the performance, the chance of such change missed during code review on method final is much lower.